src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2909,6 +2910,137 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+            If resolved <> SpecialType.None Then
+                ' Quick access table strangely maps `string Like string` to the `string` return type. remap it to 'bool'
+                ' here as that's what the operator actually is.
+                '
+                ' Similarly, the relations table doesn't include useful info.  it always has the original type,
+                ' not the expected 'bool' return type.
+                If resolved <> SpecialType.System_Object Then
+                    If opInfo.BinaryOperatorKind = BinaryOperatorKind.Equals OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.NotEquals OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.LessThanOrEqual OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.GreaterThanOrEqual OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.LessThan OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.GreaterThan OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.Like Then
+
+                        resolved = SpecialType.System_Boolean
+                    End If
+                End If
+
+                If returnType.SpecialType = resolved Then
+                    Return
+                End If
+            End If
+
+            Throw New ArgumentException($"Unsupported built-in binary operator: {returnType.ToDisplayString()} operator {name}({leftType.ToDisplayString()}, {rightType.ToDisplayString()})")
+        End Sub
+
+        Protected Overrides Function CommonCreateBuiltinOperator(
+                name As String,
+                returnType As ITypeSymbol,
+                operandType As ITypeSymbol) As IMethodSymbol
+
+            Dim vbReturnType = returnType.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(returnType))
+            Dim vbOperandType = returnType.EnsureVbSymbolOrNothing(Of NamedTypeSymbol)(NameOf(operandType))
+
+            Dim nameToCheck = If(SynthesizedIntrinsicOperatorSymbol.IsCheckedUnaryOperator(name), WellKnownMemberNames.UnaryNegationOperatorName, name)
+
+            Dim opInfo = OverloadResolution.GetOperatorInfo(nameToCheck)
+            If Not opInfo.IsUnary Then
+                Throw New ArgumentException($"Illegal operator name '{name}'", NameOf(name))
+            End If
+
+            CheckUnaryBuiltInOperator(name, vbReturnType, vbOperandType, opInfo)
+
+            Return New SynthesizedIntrinsicOperatorSymbol(vbOperandType, name, vbReturnType)
+        End Function
+
+        Private Shared Sub CheckUnaryBuiltInOperator(
+                name As String,
+                returnType As TypeSymbol,
+                operandType As NamedTypeSymbol,
+                opInfo As OverloadResolution.OperatorInfo)
+            ' 
+            If returnType.IsErrorType() OrElse operandType.IsErrorType() Then
+                Return
```

Comment:
>Return [](http://example.com/codeflow?start=16&length=6)

I think that operadndType with known SpecialType should still pass the tests below. We know what that type is.